src/test/graphql/queries/item.integration.ts

```diff
@@ -192,4 +232,83 @@ describe('item', () => {
     );
     expect(entities[0].savedItem).toBeNull;
   });
+  it('resolves more than one savedItem by itemId from multiple entities', async () => {
+    const expected = [
+      {
+        itemId: '1',
+        savedItem: {
+          url: 'https://www.youtube.com/watch?v=aWJ_7akYFhg',
+          isFavorite: true,
+          isArchived: false,
+          id: '1',
+        },
+      },
+      {
+        itemId: '999',
+        savedItem: {
+          url: 'https://www.youtube.com/watch?v=OZaL86RDGIU',
+          isFavorite: false,
+          isArchived: true,
+          id: '999',
+        },
+      },
+    ];
+    const variables = {
+      userId: '1',
+      itemId1: '1',
+      itemId2: '999',
+    };
+
+    const res = await request(app).post(url).set(headers).send({
+      query: GET_TWO_SAVED_ITEMS_ITEMID,
+      variables,
+    });
+    expect(res.body.errors).toBeUndefined;
+    expect(res.body.data).not.toBeUndefined;
+    const entities = res.body.data._entities;
+    expect(entities.length).toEqual(2);
+    expect(entities).toEqual(expect.arrayContaining(expected));
+  });
+  it('resolves savedItem by itemId field from entity representation', async () => {
+    const expected = {
+      itemId: '1',
+      savedItem: {
+        url: 'https://www.youtube.com/watch?v=aWJ_7akYFhg',
+        isFavorite: true,
+        isArchived: false,
+        id: '1',
+      },
+    };
+    const variables = {
+      userId: '1',
+      itemId: '1',
+    };
+
+    const res = await request(app).post(url).set(headers).send({
+      query: GET_SAVED_ITEM_ITEMID,
+      variables,
+    });
+    expect(res.body.errors).toBeUndefined;
+    expect(res.body.data).not.toBeUndefined;
+    const entities = res.body.data._entities;
+    expect(entities.length).toEqual(1);
+    expect(entities[0]).toEqual(expected);
+  });
+  it('returns null if the save by itemId does not exist', async () => {
```

Comment:
ah good catch, `itemId` should be 2 in the second test case, as it doesn't belong to `userId` 1.

Reply:
but how was it passing when `itemId` was `1`??

Reply:
`expect(entities[0].savedItem).toBeNull` actually was never triggered, because it should be `expect(entities[0].savedItem).toBeNull()`. So that is why it passed by fluke. If `itemId` is 1 in this test case, `expect(entities[0].savedItem).toBeNull()` is gonna fail as there is actually a `savedItem` with `itemId` 1 for `userId` 1.